### PR TITLE
Issue #29 (custom attribute parsers)

### LINKED ISSUE #29: Model class must have a custom attribute parser
The Model Class must have a custom attribute parser method.

```python

class UserModel(MinosModel):

     name: str

     def parse_name(self, name: str) -> str:
            return name.lower()

```

the format name must be

```python

parse_{variable name}

```

the method must return the parseed parameter value and this value would overlap the default attribute value

### PR DESCRIPTION
**Pull request by garciparedes**
Resolves #29

### CODE REVIEW

File: minos/common/model/fields.py [hunk 1]
```diff
@@ -41,16 +42,35 @@ def type(self) -> t.Type:
...
+        :param data: new value.
+        :return: This method does not return anything.
         """
-        check if the value is correct
-        """
+        if self._parser is not None:
+            try:
+                data = self.parser(data)
+            except Exception as exc:
+                log.warning(f"Raised {repr(exc)} while parsing {repr(self.name)} field with {repr(data)} value.")
```
Comment (garciparedes) [inline minos/common/model/fields.py:72]: Hi @andrea-mucci, should this be the expected behavior when an exception is raised while calling the corresponding parser? I mean, ignoring it and trying to continue without parsing, or should we propagate the exception?

File: tests/modelClasses.py [hunk 2]
```diff
@@ -39,6 +60,15 @@ class Customer(User):
     is_admin: Optional[bool]
     lists: Optional[list[int]]
 
+    @staticmethod
+    def parse_name(name: str) -> str:
```
Comment (garciparedes) [inline tests/modelClasses.py:64]: Hi @andrea-mucci, Another topic I want to comment with you is that for technical simplicity I imposed that parsing methods must be static (or class) methods. 

The reason why I did that is to ensure that only a single parameter is passed (and ease the internal logic),  but I want to ask you if this approach is right according to the future use cases of this feature. I mean, with the current strategy we couldn't get any other attribute of the current class instance.

File: minos/common/model/fields.py (hunk 1 above)
Comment (andrea-mucci) [reply minos/common/model/fields.py:72]: The Exception should be propagated.

File: tests/modelClasses.py (hunk 2 above)
Comment (andrea-mucci) [reply tests/modelClasses.py:64]: For the parser i suppose that only one parameter must be parsed, this is not the same for validation process.

During the daily we have to expose some use cases and check if is the right way for the parser to get only one parameter

Comment (garciparedes) [reply tests/modelClasses.py:64]: Okay @andrea-mucci! I'll proceed to apply the changes as we discussed during the daily meeting. 🙂

**Review by andrea-mucci — APPROVED** [review]

**Pull request merged**